### ISSUE TITLE
Figaro for Mac OS X needs to be 64-bit

### ISSUE THREAD

**Issue by russellcameronthomas** [p2t2/figaro#752]
I just installed the latest version of Figaro on Mac OS X, and I get the following warning during the installation process:

"Figaro" is not optimized for your Mac and needs to be updated.  This app will not work on future versions of macOS and it needs to be updated to improve compatibility.  Please contact the developer."

This web page: https://support.apple.com/en-us/HT208436 says the problem is that Figaro for Mac is a 32-bit application, and needs to be converted to 64-bit.

My OS: 10.13.6 (17G14019) (High Sierra)

I assume this requires recompiling from source code, plus resolving any library conflicts.  I assume this is possible because there is 64-bit Figaro for Linux.

I have never attempted this myself, otherwise I would do it.

Context: I am evaluating Figaro for use in a commercial software system.  If this issue isn't resolved, then we will be forced to use a different solution.

**Comment by apfeffer** [comment]
Hello,

Thanks for letting us know about this. I am cc’ing Mike Reposa, who might be able to help. Mike, do you have any suggestions?

Thanks,

Avi

**Comment by russellcameronthomas** [comment]
Any progress on this issue?

**Comment by mreposa** [comment]
Yes, but I'm still looking into it.... The 64-bit Linux version is the same library that gets installed on MacOS, because Figaro is built using a platform neutral language (Scala). So your OS may be complaining about the Figaro installer program itself (which may be a 32-bit MacOS program, I can't say for sure). Considering this is a warning, can you verify the final installation to see if the Figaro JAR file made it to your disk?

**Comment by russellcameronthomas** [comment]
I will check to see if the JAR file is on the disk.  If the complaint is about the installation program, as you suspect, then that is not a problem for us.  Also we run a mixed shop (Mac, Windows, Linux), so it's good to know about the commonality of the 64-bit library.
